src/track/beatutils.cpp

```diff
@@ -12,37 +12,314 @@
+                lastTempoChage < tempoChanges.size();
+                lastTempoChage++) {
+
+        int beatStart = tempoChanges[lastTempoChage -1];
+        // when adding the beats in a new tempo in the previous iteration
+        //  we might have already added beats at this positions so skip then...
+        if (fixedBeats.size() > 0) {
+            while (rawBeats[beatStart] <= fixedBeats.last()) {
+                beatStart += 1;
+                if (beatStart == rawBeats.size()) {
+                    break;
+                }
+            }
+        }
+        int beatEnd = tempoChanges[lastTempoChage];
+        // in this case we have already considered all these beats..
+        if (beatStart >= beatEnd) {
+            continue;
+        }
+        int partLenght = beatEnd - beatStart;
+        double leftRighDiff = 1.0;
+        // here we detect if the segment has a constant tempo or not
+        if (partLenght >= kBeatsToCountTempo * 2) {
+            int middle = partLenght / 2;            
+            auto beatsAtLeft = QVector<double>::fromStdVector(std::vector<double>(
+                    rawBeats.begin() + beatStart, rawBeats.begin() + beatStart + middle));
+            auto beatsAtRight = QVector<double>::fromStdVector(std::vector<double>(
+                    rawBeats.begin() + beatStart + middle, rawBeats.begin() + beatEnd));
+            QMap<double, int> leftTempoFrequency;
+            auto temposLeft = computeWindowedBpmsAndFrequencyHistogram(
+                    beatsAtLeft, kBeatsToCountTempo, 1, sampleRate, &leftTempoFrequency);
+            QMap<double, int> rightTempoFrequency;
+            auto temposRight = computeWindowedBpmsAndFrequencyHistogram(
+                        beatsAtRight, kBeatsToCountTempo, 1, sampleRate, &rightTempoFrequency);
+            double modeAtLeft = BeatStatistics::mode(leftTempoFrequency);
+            double modeAtRight = BeatStatistics::mode(rightTempoFrequency);
+            leftRighDiff = fabs(modeAtLeft - modeAtRight);
+        }
+        // const tempo make const grid
+        if (leftRighDiff < kMaxDiffSameBpm) {
+            auto splittedAtTempoChange = QVector<double>::fromStdVector(std::vector<double>(
+                rawBeats.begin() + beatStart, rawBeats.begin() + beatEnd));
+            double bpm = calculateBpm(splittedAtTempoChange, sampleRate, minBpm, maxBpm);
+            fixedBeats << calculateFixedTempoGrid(splittedAtTempoChange, sampleRate, bpm);
+        }
+        // not const make ironed grid
+        else {
+            auto splittedAtTempoChange = QVector<double>::fromStdVector(std::vector<double>(
+                rawBeats.begin() + beatStart, rawBeats.begin() + beatEnd));
+            fixedBeats << calculateIronedBeatMap(splittedAtTempoChange, sampleRate);
+        }
+    }
+    return fixedBeats;
+}
+
+QVector<double> BeatUtils::calculateFixedTempoGrid(
+        const QVector<double> &rawbeats, int sampleRate, const double localBpm) {
```

Comment:
Normally the calculateIronedBeatMap should give quite similar results. 
Do you have a track where this is not the case?

Reply:
Similar yes, but not as good for true const tempo.
With this, we avoid any change at all when the tempo is const but the ironing will still have some changes, so I think this works better...

Reply:
Yes, fo example https://www.youtube.com/watch?v=W-nrHptw4Ow
And actually any edm track with const tempo.

Reply:
The last section in the magic track is the real issue.
Do we detect that as arrythmic section?

Please describe the issue as comment.